# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerInterceptController.cpp
# Commit: af27682c5e7f
# Full Hash: af27682c5e7f86ff17d6ed152c8010e763870bdc
# Author: ssengupta <ssengupta@mozilla.com>
# Date: 2020-05-13 03:44:05
# Regressor Bug: 1635399
# File Overlap Count: 1
# Description:
#   Bug 1635399 - Function PrincipalInfoToPrincipal now returns Result<nsCOMPtr<nsIPrincipal>, nsresult> r=ckerschb
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73868
# ==============================================================================

diff -r 86b22b1a9f7e -r af27682c5e7f dom/serviceworkers/ServiceWorkerInterceptController.cpp
--- a/dom/serviceworkers/ServiceWorkerInterceptController.cpp	Tue May 12 23:55:42 2020 +0300
+++ b/dom/serviceworkers/ServiceWorkerInterceptController.cpp	Tue May 12 19:02:05 2020 +0000
@@ -8,9 +8,11 @@
 
 #include "mozilla/BasePrincipal.h"
 #include "mozilla/StorageAccess.h"
+#include "nsCOMPtr.h"
 #include "nsContentUtils.h"
 #include "nsIChannel.h"
 #include "ServiceWorkerManager.h"
+#include "nsIPrincipal.h"
 
 namespace mozilla {
 namespace dom {
@@ -49,9 +51,10 @@
       // Get ServiceWorkerRegistrationInfo by the ServiceWorkerInfo's principal
       // and scope
       if (!*aShouldIntercept && swm) {
+        nsCOMPtr<nsIPrincipal> principal =
+            controller.ref().GetPrincipal().unwrap();
         RefPtr<ServiceWorkerRegistrationInfo> registration =
-            swm->GetRegistration(controller.ref().GetPrincipal().get(),
-                                 controller.ref().Scope());
+            swm->GetRegistration(principal, controller.ref().Scope());
         // Could not get ServiceWorkerRegistration here if unregister is
         // executed before getting here.
         if (NS_WARN_IF(!registration)) {